plugin-api/src/main/java/org/sonar/api/resources/Language.java

```diff
@@ -60,4 +72,18 @@ public interface Language {
   default boolean publishAllFiles() {
     return true;
   }
+
+  /**
+   * Provide a list of patterns following the {@link org.sonar.api.utils.WildcardPattern} syntax. Source files matching any of those patterns will
+   * be associated to this language.
+   * Pattern are considered relative: <code>["pom.xml"]</code> is equivalent to <code>["&#42;&#42;/pom.xml"]</code> <br/>
+   * The filename extension matching is case-insensitive, so declaring <code>["&#42;&#42;/&#42;Test.java"]</code> will match "FooTest.java" and
+   * "FooTest.JAVA" but <strong>not</strong> "FooTEST.java".<br/>
```

Comment:
@henryju Is this true for this option? We are describing here **filename extensions** where in fact it is about **filename patterns** - extensions have nothing to do with it. Am I missing something? IMO, if you define a file pattern like: `**/*Text.java, then  `FooTest.JAVA` must not match - at least according to `WildcardPattern` implementation.

Reply:
For consistency (and probably easier implementation), I decided to keep the same behavior as for `getFileSuffixes`. See https://github.com/SonarSource/sonar-enterprise/blob/165754f63e45a038afba2a5d8c8415ff51c2631f/sonar-plugin-api-impl/src/main/java/org/sonar/api/batch/fs/internal/PathPattern.java#L82
In fact, the javadoc is not totally exact. It is not that the extension matching is case-insensitive, but more that file extensions are changed to lowercase before being compared to the pattern.
So filename `foo/bar/Program.JAVA` would be "normalized" to `foo/bar/Program.java` and so would be matched by `**/*.java`
But pattern `**/*.JAVA` would not match anything.
I am not sure it is worth being documented, since I think this is more implementation detail, and the original intent was probably to have proper case insensitivity for file extensions, assuming `getFileSuffixes` would only provide lowercase values.